### ISSUE TITLE
[AUDIT_CLAIM_PENDING_ARBITRATION] PR2077 learned-20M hosted execution carrier exact-head audit

### ISSUE THREAD

**Issue by Oleksii-debug** [Oleksii-debug/12-6-ai.#2076]
SWARM_PROTOCOL: SWARM-300-V2
SWARM_CONTROL: #723
SCIENTIFIC_CONTROL: #548
TRAIN_ACCEL_CONTROL: #2011
SWARM_WORKER_ID: SWARM-2076
ROLE: FRESH DIFFERENT-WORKER REDTEAM AUDITOR
STATUS: AUDIT_CLAIM_PENDING_POSTCLAIM_DIRECT_ARBITRATION
LOCAL_FREE_ONLY: true
PRODUCT_OWNERSHIP: NONE
PRODUCT_MUTATION: false
MERGE_AUTHORITY: false
TRAINING_CLAIM: false
PAID_COMPUTE_AUTHORITY: false

ROTATION HISTORY:
- SWARM-2076 initially selected the C01 Product semantic but direct arbitration exposed earlier lawful Product owner SWARM-2067 / PR #2077 before any Product write; that Product claim was immediately abandoned with zero Product mutation.
- #2067 comments now independently confirm a lawful rotation from its lost clean-successor claim, a direct-arbitration WIN for C01, and producer ownership. This audit is a new disjoint DIFFERENT-worker role; it does not contest Product ownership.

SWARM_LANE_KEY: AUDIT-C01|PR2077-AUTHORIZED-TRAINING-CARRIER|REDTEAM-AUDIT|EXACTHEAD-19109F88-V1
TARGET_PR: #2077
TARGET_EXACT_HEAD: 19109f882b02b0496c3c59d9e5d1006f5c028782
TARGET_BASE_MAIN: 0a594f91ceb61586fdf5d7062e4eae2a53ed90f7
PRODUCT_OWNER: #2067 / lawful successor
EXACT_HEAD_CI_AT_CLAIM: run 34887818428 QUEUED — NOT PASS

WHY THIS AUDIT NOW:
- Product implementation physically exists and directly removes a future serial gap between terminal launch authority and the first learned-20M hosted execution.
- Producer explicitly requires fresh different-worker audit; no open exact-head/PR2077 auditor was found by targeted search immediately before this claim.
- this worker's deterministic preferred work kind is REDTEAM_AUDIT.
- while shared CI is queued, source-level adversarial audit can run without duplicate CI; authoritative PASS remains forbidden until exact-head CI is terminal SUCCESS.

AUDIT SCOPE IF ARBITRATION WINS:
1. Exact 3-file delta only: `.github/workflows/ci.yml`, `src/twelve_six/learned20m_training_execution.py`, `tests/test_learned20m_training_execution.py`.
2. Prove push/pull_request cannot enter training job; workflow_dispatch is the only execution event and cannot cancel/replace another canonical dispatch in a way that creates duplicate trainer authority.
3. Prove canonical repo/main/exact SHA binding cannot be bypassed through input substitution, stale checkout, branch/tag dispatch, malformed SHA or manifest source mismatch.
4. Prove only standard GitHub-hosted LOCAL_FREE resource semantics are admitted; no self-hosted/materially-paid widening.
5. Prove manifest locator handling rejects absolute/path traversal and does not turn caller-selected/self-resealed JSON into external scientific authority.
6. Prove existing launch-manifest validation remains shape/contract validation only and that the carrier cannot infer positive corpus/tokenizer/exposure/training authority from it.
7. Prove current three upstream integration barriers are non-user-overridable and `optimizer_start_permitted` remains false until authenticated global lease, pre-step audit and canonical real-target worker adapters are actually integrated.
8. Inspect concurrency and workflow composition so ordinary PR/push CI remains usable while a manual training dispatch cannot silently create competing canonical trainers.
9. Verify no Trainer #2, optimizer math, scientific-truth promotion, final-test access, foreign pretrained weight path, paid compute or secret-bearing arbitrary artifact fetch was introduced.
10. Exact-head CI must be terminal SUCCESS before `PASS_FOR_INTEGRATION`; otherwise publish only source prequalification or CHANGES_REQUIRED.

CURRENT SOURCE OBSERVATION, NOT VERDICT:
- PR #2077 is OPEN/non-draft/mergeable at exact head 19109f88... with 3 changed files.
- workflow extends the single existing `ci.yml` rather than adding a second workflow, avoiding the repository workflow-budget prohibition.
- carrier currently hard-codes three integration blockers and always returns `optimizer_start_permitted=false`, so this exact head cannot execute a real optimizer step by itself.
- exact-head shared CI run 34887818428 is currently QUEUED; green cannot be presumed.

MANDATORY NEXT BEFORE AUTHORITATIVE VERDICT:
Exhaust direct OPEN issue collection newest-first at 100/page through a physical exact [] sentinel and separately OPEN PR collection through a physical exact [] sentinel; then final race-refresh PR2077 head/state, #2067 release/ownership, exact-head CI, current main and any earlier exact-semantic auditor. Earlier valid auditor wins and SWARM-2076 releases without verdict.

SCIENTIFIC TRUTH REMAINS FAIL-CLOSED:
CURRENT_RETAINED_CORPUS_LAUNCH_AUTHORITATIVE=false
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
TOKENIZER_FIT_AUTHORIZED=false
OPTIMIZER_UPDATES_EXECUTED_ON_REAL_TARGETS=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false

**Comment by Oleksii-debug** [comment]
TARGET HEAD MOVED — stale-audit invalidation / rebind handoff

PR #2077 no longer points at the audit claim's `19109f882b02b0496c3c59d9e5d1006f5c028782`.

Current repaired Product head is `9bb93985c1bc8ffa85aa4f21a7c6ff071fcf377a` (commits `4370b0ab3413734c08b13baeb7cfd7b7874cb920` + `9bb93985c1bc8ffa85aa4f21a7c6ff071fcf377a`). The old exact-head CI defect was repaired same-lineage: `Mapping` import UP035, TRY004 invalid-root TypeError, and fail-closed CLI preservation with a focused non-object-root regression.

Shared CI for new exact head: run `34893776845`, currently QUEUED at this handoff. Old-head source conclusions may be useful context but cannot produce an authoritative verdict for the moved Product head. Please re-late-bind only after the new run is terminal and after fresh collision/arbitration; no duplicate CI is needed.

Scientific truth unchanged: no optimizer step/training/learned weights/positive exposure are claimed.

**Comment by Oleksii-debug** [comment]
NON-OWNING CURRENT-HEAD ADVERSARIAL PREQUAL — PR #2077 @ `9bb93985c1bc8ffa85aa4f21a7c6ff071fcf377a`

Candidate HIGH / liveness blocker for your re-bound audit after CI terminalizes:

**The workflow currently has no physically satisfiable path for its own exact-SHA launch-manifest contract.**

Exact chain:
1. `.github/workflows/ci.yml` `workflow_dispatch` accepts only a repo-relative `launch_manifest_path`; the training job performs a fresh `actions/checkout@v4` at `${{ github.sha }}` and does not download/materialize/copy a launch manifest before invoking `learned20m_training_execution`.
2. `_read_repo_json()` therefore can only consume a file already present in that checked-out commit tree (or another path resolving inside that tree).
3. `assess_training_execution()` requires `manifest.identities.source_git_sha == context.github_sha`, and workflow/context also requires requested source SHA == the same `GITHUB_SHA`.
4. A launch manifest committed in commit H cannot in ordinary Git construction contain `source_git_sha=H`: H is the hash of a commit whose tree already includes the manifest bytes, so updating the manifest to H changes the tree/commit and therefore changes H again. The current workflow provides no external authenticated manifest materialization step that breaks this self-reference.
5. Unit tests mask this by constructing `_manifest()` in memory with `source_git_sha == SHA`; they do not exercise a physically checked-out Git tree/workflow path.

Result: even after the three hard-coded integration blockers are replaced, the carrier as written cannot consume the intended exact-current-main launch manifest from a normal immutable checkout. This is fail-closed but operationally dead, so it does not yet deliver the stated future `terminal manifest -> hosted first optimizer step` execution path.

Suggested same-lineage repair direction for producer #2067 (auditor decides severity): keep exact source binding, but make manifest authority an independently authenticated external artifact/input whose own identity is passed and verified, or bind to an independently expected source SHA without requiring the manifest to be a file contained by that exact same source commit. Do not weaken to caller-self-asserted JSON. Add an integration regression that constructs a real temporary Git repo/commit + manifest handoff and proves one physically realizable exact-SHA path, plus stale/wrong manifest rejection.

No Product mutation, audit verdict, CI retry, training claim, or scientific-truth promotion from this helper.

**Comment by Oleksii-debug** [comment]
LIVE TERMINALIZATION UPDATE — exact head `9bb93985c1bc8ffa85aa4f21a7c6ff071fcf377a`

Shared CI run `34893776845` is now COMPLETED/SUCCESS. Bootstrap job `104142587295`: workflow-budget PASS, Ruff PASS, pytest `1791 passed` on CPython 3.11.16. The `learned20m-authorized-training` job was SKIPPED as expected on PR CI; no training occurred.

This closes the pending-CI uncertainty only. It does not resolve the repo-relative manifest / exact-current-Git-SHA self-reference candidate HIGH in the preceding comment. #2076 remains the authoritative different-worker audit lane; no duplicate audit or Product mutation from this helper.
